test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua

```diff
@@ -0,0 +1,75 @@
+local net = require('net.box')
+local server = require('luatest.server')
+local t = require('luatest')
+
+local g = t.group()
+
+g.before_all(function()
+    g.server = server:new({alias = 'master'})
+    g.server:start()
+    g.server:exec(function()
+        box.schema.create_space('test', {
+            format = {
+                {name = "id", type = "unsigned"},
+                {name = "name", type = "string"},
+            },
+        })
+
+        box.space.test:create_index('primary', {
+            type = 'tree',
+            unique = true,
+            parts = {1, 'unsigned'},
+        })
+
+        box.space.test:create_index('secondary', {
+            id = 5,
+            type = 'tree',
+            unique = true,
+            parts = {2, 'string'},
+        })
+    end)
+end)
+
+g.after_all(function()
+    g.server:drop()
+end)
+
+g.before_each(function()
+    -- fetch schema = false to resolve index on server instead of client
+    g.conn = net.connect(g.server.net_box_uri, {fetch_schema = false})
+    t.assert_is_not(g.conn, nil)
+
+    g.server:exec(function()
+        box.space.test:truncate()
+    end)
+end)
+
+g.after_each(function()
+    g.conn:close()
+end)
+
+g.test_index_select = function()
+    local space = g.conn.space.test
+    t.assert_is_not(space, nil)
+
+    space:insert({1, 'F'})
+
+    local res1 = space.index.secondary:select('F')
+    local res2 = space.index[5]:select('F')
+    t.assert_equals(res1, res2)
+end
+
+g.test_index_delete = function()
+    local space = g.conn.space.test
+    t.assert_is_not(space, nil)
+
+    space:insert({2, 'A'})
+    space:insert({4, 'B'})
+    space:insert({8, 'C'})
```

Comment:
What's the point of inserting other values here? Please keep your tests concise and straight to the point, so that a reviewer and future reader of your test does not spend time guessing why this is needed.